iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/SectionedNameStrictMatchVoter.java

```diff
@@ -17,35 +18,48 @@
     
     private static final long serialVersionUID = 1L;
     
+    private Function<AffMatchOrganization, List<String>> getOrgNamesFunction = new GetOrgNameFunction();
+    
     
+    
+        
     //------------------------ LOGIC --------------------------
+     
+    
     
     /**
-     * Returns {@link AffMatchAffiliation#getOrganizationName()} of the passed affiliation.
+     * Returns true if any of the affOrgNameSections is equal to orgNameSection.
      */
     @Override
-    protected String getAffiliationName(AffMatchAffiliation affiliation) {
-        return affiliation.getOrganizationName();
+    protected boolean containsOrgSection(List<String> affOrgNameSections, String orgNameSection) {
+        return affOrgNameSections.contains(orgNameSection);
     }
+
     
+    @Override
+    protected List<String> getOrganizationNames(AffMatchOrganization organization) {
```

Comment:
Looking at different `Voters` implementations utilizing `getOrgNamesFunction` field I guess we could introduce another `orgNamesFunction`-aware abstract class in hierarchy including:
- `getOrgNamesFunction` field
- `setGetOrgNamesFunction` method
- `getOrganizationNames` method

this way we would have it implemented in single place.